File: CoreTransformer/TransitionSizeDistributionCollection.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Imports System.Collections.ObjectModel

''' <summary>
''' The transition size distribution collection class
''' </summary>
''' <remarks></remarks>
Friend Class TransitionSizeDistributionCollection
    Inherits KeyedCollection(Of Integer, TransitionSizeDistribution)

    ''' <summary>
    ''' Gets the key for the specified item
    ''' </summary>
    ''' <param name="item"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Protected Overrides Function GetKeyForItem(ByVal item As TransitionSizeDistribution) As Integer
        Return item.TransitionSizeDistributionId
    End Function

End Class

File: src/Runtime/TransitionSizeDistributionCollection.cs
﻿// A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class TransitionSizeDistributionCollection : KeyedCollection<int, TransitionSizeDistribution>
    {
        protected override int GetKeyForItem(TransitionSizeDistribution item)
        {
            return item.TransitionSizeDistributionId;
        }
    }
}